pkg/cmd/agent/agent.go

```diff
@@ -3,20 +3,49 @@
 package main
 
 import (
+	"context"
 	"net/http"
 	"runtime/debug"
 	"syscall"
 
 	"github.com/go-chi/chi/v5"
 	"github.com/go-chi/render"
 	"github.com/pkg/errors"
+	"github.com/scylladb/go-log"
 	"github.com/scylladb/scylla-manager/v3/pkg/config/agent"
+	"github.com/scylladb/scylla-manager/v3/pkg/util/cpuset"
+	"github.com/scylladb/scylla-manager/v3/swagger/gen/agent/models"
 )
 
-func newAgentHandler(c agent.Config, rclone http.Handler) *chi.Mux {
+func newAgentHandler(c agent.Config, rclone http.Handler, logger log.Logger) *chi.Mux {
 	m := chi.NewMux()
+	ctx := context.Background()
 
 	m.Get("/node_info", newNodeInfoHandler(c).getNodeInfo)
+	m.Get("/pin_cpu", func(writer http.ResponseWriter, request *http.Request) {
+		cpus, err := cpuset.SchedGetAffinity()
+		if err != nil {
+			render.Status(request, http.StatusInternalServerError)
+			render.Respond(writer, request, err)
+		}
+		casted := make([]int64, len(cpus))
+		for i := range cpus {
+			casted[i] = int64(cpus[i])
+		}
+		render.Respond(writer, request, &models.Cpus{CPU: casted})
+	})
+	m.Post("/pin_cpu", func(writer http.ResponseWriter, request *http.Request) {
+		if err := findAndPinCPUs(ctx, c, logger); err != nil {
```

Comment:
By doing it, you pass the context created during handler creation.
It makes no sense, as the context here is the request.
There is a context available in `request *http.Reqest`, please use this one.
`request.Context()`